src/watertap_contrib/seto/property_models/basic_water_properties.py

```diff
@@ -0,0 +1,397 @@
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of of logging
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+
+        if flags is None:
+            return
+
+        # Unfix state variables
+        revert_state_vars(self, flags)
+        init_log.info("State Released.")
+
+
+@declare_process_block_class("BasicWaterStateBlock", block_class=_BasicWaterStateBlock)
+class BasicWaterStateBlockData(StateBlockData):
+    """
+    General purpose StateBlock for Zero-Order unit models.
+    """
+
+    def build(self):
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.flow_vol = Var(
+            initialize=1,
+            domain=PositiveReals,
+            doc="Volumetric flow rate",
+            units=pyunits.m**3 / pyunits.s,
+        )
+
+        self.conc_mass_comp = Var(
+            self.params.solute_set,
+            initialize=1,
+            domain=PositiveReals,
+            doc="Mass concentration of each solute",
+            units=pyunits.kg / pyunits.m**3,
+        )
+
+    # Other properties
+    def _flow_mass_comp(self):
+
+        self.flow_mass_comp = Var(
+            self.params.component_list,
+            initialize=1,
+            domain=PositiveReals,
+            doc="Mass flowrate of each component",
+            units=pyunits.kg / pyunits.s,
+        )
+
+        def rule_flow_mass_comp(b, j):
+            if j == "H2O":
+                return b.flow_mass_comp[j] == b.flow_vol * b.dens_mass
+            else:
+                return b.flow_mass_comp[j] == b.flow_vol * b.conc_mass_comp[j]
+
+        self.eq_flow_mass_comp = Constraint(
+            self.params.component_list, rule=rule_flow_mass_comp
+        )
+
+    def _temperature(self):
+        self.temperature = Var(
+            initialize=298.15,
+            bounds=(273.15, 373.15),
+            units=pyunits.K,
+            doc="Temperature",
+        )
+
+    def _pressure(self):
+        self.pressure = Var(
+            initialize=101325,
+            bounds=(1e5, None),
+            units=pyunits.Pa,
+            doc="Pressure",
+        )
+
+    def _dens_mass(self):
+        add_object_reference(self, "dens_mass", self.params.dens_mass)
+
+    def _visc_d(self):
+        add_object_reference(self, "visc_d", self.params.visc_d)
+
+    def get_material_flow_terms(self, j):
+        return self.flow_mass_comp[j]
+
+    def get_enthalpy_flow_terms(self, p):
+        raise NotImplementedError
+
+    def get_material_density_terms(self, j):
+        if j == "H2O":
+            return self.dens_mass
+        else:
+            return self.conc_mass_comp[j]
+
+    def get_energy_density_terms(self, p):
+        raise NotImplementedError
+
+    def default_material_balance_type(self):
+        return MaterialBalanceType.componentTotal
```

Comment:
I wonder if this could cause error if someone tries to add material balance, because we assume the same density for the solution and solvent. (could be wrong)

Reply:
That's a good point. @adam-a-a should it be ``MaterialBalance.none`` (if that exists) since this is meant to be basic?

Reply:
MaterialBalance.none does exist but I don't know that this is an issue just yet.